### DISCUSSION TITLE
Android tablet trackpad malfunctions when connecting to computer remotely

### DISCUSSION THREAD

**Discussion by Kyochilian** [rustdesk/rustdesk#10131] [Feature Request]
### Bug Description

Hi developer, thanks for your time!
When connecting my Android tablet to a Windows device, the official Bluetooth keyboard keys function normally, but the trackpad is unusable.I'm not sure if this is a bug, but the trackpad really doesn't allow mouse movement：

### How to Reproduce

Device:
Andriod Tablet:Xiaomi Pad 6s Pro,the offical keyboard with trackpad;
Windows:Dell Inspiron with Windows 10;
Own server to connect;
Xiaomi Pad 6s Pro -> Windows 10, the trackpad didn't working.

### Expected Behavior

When I move my finger on the trackpad, the mouse moves accordingly.

### Operating system(s) on local (controlling) side and remote (controlled) side

Andriod Tablet -> Windows 10

### RustDesk Version(s) on local (controlling) side and remote (controlled) side

1.3.2 -> 1.3.2

### Screenshots

![image](https://github.com/user-attachments/assets/3441d2be-3e29-42ca-80c6-d2dd4319f458)

### Additional Context

I'm not sure if this is a bug or an issue with the keyboard itself.
Thanks for your time again!

**Comment by haoruizhou** [comment]
Same issue with Huawei MatePad pro trying to control Mac

**Comment by fufesou** [comment]
@Kyochilian @haoruizhou Could you please test this build https://github.com/fufesou/rustdesk/releases/tag/test-stylus-trackpad ?

**Comment by haoruizhou** [comment]
@fufesou 
Thank you for the build.
I downloaded the [rustdesk-1.4.0-aarch64.apk](https://github.com/fufesou/rustdesk/releases/download/test-stylus-trackpad/rustdesk-1.4.0-aarch64.apk) build on a Huawei MatePad Pro 13.2 (MRO-W00), paired with its official Star Flash keyboard. The host device is a MacBook Pro 16, running macOS Sequoia 15.5. Only left click is registered by RustDesk, both under touch and mouse mode. The pointer of the tablet remains functional to interact with rest of the UI of RustDesk, but not the remote machine.

**Reply by fufesou** [reply]
Thanks a lot for your test.

But I can't understand clearly.

> Only left click is registered by RustDesk, both under touch and mouse mode.

Do you mean you can use left click with this build  https://github.com/fufesou/rustdesk/releases/tag/test-stylus-trackpad , but can't with the official build?

**Reply by haoruizhou** [reply]
My apologies. I just tested again and noticed that a left click on the trackpad is NOT registered. I probably accidentally touched the screen in the previous test.

**Reply by haoruizhou** [reply]
https://github.com/user-attachments/assets/8672ad86-82e1-4caf-9f5e-6b343f569184

I find it hard to explain what exactly is going on:
The top (system status bar UI) and bottom (RustDesk UI) can be interacted with the mouse. However, if I left click when my tablet pointer is within the field of MacBook's screen, the mouse get "captured" and "stuck" into the MacBook's screen until I keep moving my finger on the trackpad to move it down or up to the top or bottom UIs.

**Reply by fufesou** [reply]
Thanks a lot. The trackpad events are still not handled by RustDesk.
I'll look for the reason again.

**Reply by CottonWhiteSheep** [reply]
I'm facing the same issue on Xiaomi Pad. I can move the cursor of the remote device using two fingers, but whenever I perform a click, the remove cursor jumps to the position where the Android device's system cursor is located.

**Reply by fufesou** [reply]
@CottonWhiteSheep Hi, could you please test this build https://github.com/fufesou/rustdesk/releases/tag/test-stylus-trackpad-2 ?

This build does not fix the issue, but it prints some logs on the screen. Please use your trackpad to operate, and record a video. Thanks!

> I can move the cursor of the remote device using two fingers

The cusor can be moved using two fingers. Maybe you moved the canvas.

**Reply by CottonWhiteSheep** [reply]
https://github.com/user-attachments/assets/4a91e692-b093-47a8-8794-ccb59d83fc8c

For this one first I move it with two finger at the same time and the cursor will follow it, but as I click the cursor will jump to the Android device cursor position.

__________

https://github.com/user-attachments/assets/f9c4c179-82dd-46c8-a3c7-65c6acd6cc6a

For this I move it using one figure but only the system cursor is moving, but if if I hold down the click and drag then both will move together

**Reply by fufesou** [reply]
@haoruizhou Could you test this and upload a video? Thanks! I'm not sure if the event kinds are the same. The video above is touch, but maybe it is trackpad in your test.

**Comment by fufesou** [comment]
@haoruizhou Could you please test this build https://github.com/fufesou/rustdesk/releases/tag/test-stylus-trackpad-2 ?

This build does not fix the issue, but it prints some logs on the screen. Please use your trackpad to operate, and record a video. Thanks!

**Reply by haoruizhou** [reply]
Hi sorry it's been a long day will get on it soon.

**Reply by haoruizhou** [reply]
https://streamable.com/xny2j5
Please see this video. It seems like sometimes right click is sent to the host device.

**Comment by fufesou** [comment]
@haoruizhou @CottonWhiteSheep Hi, could you please test this build? https://github.com/fufesou/rustdesk/releases/tag/test-stylus-trackpad-2

It may fix the trackpad issue.

It also add some debug prints on the screen. Please also record a video.

Thanks!

**Reply by CottonWhiteSheep** [reply]
The cursor is still not moving, here's the video, I also include tap and double finger tap at the end 

https://github.com/user-attachments/assets/bbf83f00-2ac4-42c0-aeca-e14b23f891f5



Thank you so much

**Reply by fufesou** [reply]
@CottonWhiteSheep Thanks for your tests and feedback.

Could you please test this build again? I've updated it. https://github.com/fufesou/rustdesk/releases/tag/test-stylus-trackpad-2

Maybe we can communicate directly via email. `bGlubG9uZzEyNjZAZ21haWwuY29t` (base64 encoded).

**Reply by CottonWhiteSheep** [reply]
Hello, I have just send you an email with the video attached, thank you so much.

**Comment by eitchtee** [comment]
I've been having similar issues using a Galaxy Tab S10 FE and a non-official bluetooth keyboard using the latest Rustdesk nightly due to stylus support.

Using touch mode, only the left click is recognized by Windows, middle click doesn't work (which seems to be an Android thing), and right click and two finger scrolling doesn't seem to be recognized.

Anydesk has similar issues, with (unfortunately) only Chrome Remote Desktop being able to successfully parse all commands from the trackpad besides the middle click, with the cursor disappearing whenever I scroll. Really weird stuff.

**Comment by MarkGor** [comment]
Xiaomi Pad8Pro 同样问题，最新版本客户端1.4.4 触摸板识别不了。

**Comment by metapsych** [comment]
Same issue with Xiomi pad 7. Connecting external mouse via otg is working fine. Wonder why trackpad has this issue.
